src/main/java/jenkins/plugins/logstash/LogstashInstallation.java

```diff
@@ -95,14 +95,10 @@ public String getDisplayName() {
     }
 
     /**
-     * @see IndexerDaoFactory#getInstance(IndexerType, String, int, String, String)
+     * @see IndexerDaoFactory#getInstance(IndexerType, String, int, String, String, String)
      */
     public LogstashIndexerDao getIndexerDao() throws InstantiationException {
-      if (StringUtils.isEmpty(host)) {
-        return null;
-      }
-
-      return IndexerDaoFactory.getInstance(type, host, port, key, username, password);
+      return IndexerDaoFactory.getInstance(type, host, (port == null ? -1 : port.intValue()), key, username, password);
```

Comment:
The port is represented as `Integer` in the Descriptor and `int` in the DAO. :cry: This handles the null case.